pkgs/misc/vim-plugins/update.py

```diff
@@ -515,23 +516,29 @@ def is_unexpectedly_dirty(self) -> bool:
         return self.repo.is_dirty() and not self.allow_dirty
 
 
-def update_plugins(input_file: str, outfile: str, cache: Cache) -> Dict:
-    plugin_names = load_plugin_spec(input_file)
-    prefetch_with_cache = functools.partial(prefetch, cache=cache)
+class PluginUpdater:
+    def __init__(self, input_file: str, outfile: str):
+        self.input_file: str = input_file
+        self.outfile: str = outfile
+        self.cache: Cache = Cache(get_current_plugins())
+        self.prefetch = functools.partial(prefetch, cache=self.cache)
 
-    try:
-        # synchronous variant for debugging
-        # results = list(map(prefetch_with_cache, plugin_names))
-        pool = Pool(processes=30)
-        results = pool.map(prefetch_with_cache, plugin_names)
-    finally:
-        cache.store()
+    def __call__(self) -> Dict:
```

Comment:
I'd prefer to give this function a name, such as `updater.update()`, to be more explicit (`PluginUpdater` doesn't seem "callable" to me).